include/miroil/miroil/input_device_observer.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2016-2021 Canonical, Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+ * SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef MIROIL_INPUT_DEVICE_OBSERVER_H
+#define MIROIL_INPUT_DEVICE_OBSERVER_H
+#include <mir_toolkit/mir_input_device_types.h>
+#include <memory>
+#include <string>
+
+namespace mir { namespace input { class Device; } }
+
+namespace miroil
+{
+class InputDeviceObserver
+{
+public:
+    virtual ~InputDeviceObserver();
+    
+    void             apply_keymap(const std::shared_ptr<mir::input::Device> &device, const std::string & layout, const std::string & variant);
```

Comment:
Mir style has `const` "on the right" and "&" or "*" with the type. Vis:
```c++
    void             apply_keymap(std::shared_ptr<mir::input::Device> const& device, std::string const& layout, std::string const& variant);
```

Reply:
Okey, I'll split the class in two and move the const.